src/components/dialogmoteunntak/SettDialogmoteunntakSkjema.tsx

```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import { Form } from "react-final-form";
+import { Link, Redirect } from "react-router-dom";
+import styled from "styled-components";
+import Panel from "nav-frontend-paneler";
+import { AlertStripeInfo } from "nav-frontend-alertstriper";
+import { moteoversiktRoutePath } from "@/routers/AppRouter";
+import { useDialogmotekandidat } from "@/data/dialogmotekandidat/dialogmotekandidatQueryHooks";
+import {
+  NewUnntakDTO,
+  UnntakArsak,
+} from "@/data/dialogmotekandidat/types/dialogmoteunntakTypes";
+import SettDialogmoteunntakVelgArsak, {
+  settDialogmoteunntakVelgArsakFieldName,
+} from "@/components/dialogmoteunntak/SettDialogmoteunntakVelgArsak";
+import { TrackedHovedknapp } from "@/components/buttons/TrackedHovedknapp";
+import { TrackedFlatknapp } from "@/components/buttons/TrackedFlatknapp";
+import { FlexGapSize, FlexRow } from "@/components/Layout";
+import { useValgtPersonident } from "@/hooks/useValgtBruker";
+import { useSettDialogmoteunntak } from "@/data/dialogmotekandidat/useSettDialogmoteunntak";
+import { SkjemaInnsendingFeil } from "@/components/SkjemaInnsendingFeil";
+
+const texts = {
+  noBrev: "Det blir ikke sendt ut brev ved unntak.",
+  send: "Sett unntak",
+  avbryt: "Avbryt",
+};
+
+const StyledPanel = styled(Panel)`
+  margin-bottom: 2em;
+  padding: 2em;
+`;
+
+interface SettDialogmoteunntakSkjemaValues {
+  arsak: UnntakArsak;
+}
+
+const SettDialogmoteunntakSkjema = () => {
+  const personIdent = useValgtPersonident();
+  const { isKandidat } = useDialogmotekandidat();
+  const settDialogmoteunntak = useSettDialogmoteunntak();
+
+  if (!isKandidat || settDialogmoteunntak.isSuccess) {
+    return <Redirect to={moteoversiktRoutePath} />;
+  }
+
+  const validate = (values: Partial<SettDialogmoteunntakSkjemaValues>) => {
+    return (
+      (!values.arsak && {
+        [settDialogmoteunntakVelgArsakFieldName]: "Vennligst angi årsak",
+      }) ||
+      {}
+    );
+  };
+
+  const submit = (values: SettDialogmoteunntakSkjemaValues) => {
+    if (values.arsak) {
```

Comment:
Skal ikke være nødvendig med denne sjekken når validate validerer dette (submit kjører ikke hvis validate returnerer feil)

Reply:
Ikke slik som koden er nå se jeg, men det er fordi jeg har dette:
```interface DialogmoteunntakSkjemaValues {
  arsak: UnntakArsak;
}```
Men her burde det vel heller være: `  arsak: UnntakArsak;`? For det er jo kanskje ikke definert og da må man i så fall ha denne sjekken for at det skal gå med typescript.

Reply:
Tror vi har brukt interface *SkjemaValues for ønskede verdier (ikke valgfrie) i de andre skjemaene, sånn du har gjort her. Og så bruke Partial<*SkjemaValues> som argument til validate for der kan det være at de ikke er definert.

Reply:
Da er det mulig jeg har mistforstått Partial, trodde den var for å definere et delsett av en type, ikke at man kan bytte på ?-tegnet i typene.

Reply:
Partial brukes for å gjøre alle properties i en type optional (ref: https://www.typescriptlang.org/docs/handbook/utility-types.html. Ala:
```
type B = {
  foo: string;
  bar: boolean;
};
type A = Partial<B>; 
```
hvor da `type A = { foo?: string | undefined; bar?: boolean | undefined; }`

Reply:
Kan du ikke da la DialogmoteunntakSkjemaValues være:
interface DialogmoteunntakSkjemaValues {
  arsak: UnntakArsak;
}
så kan du droppe if-sjekken i submit og la validate bruke Partial (sånn det er nå)

Reply:
Skjønner, da gjør jeg det.